icu4c/source/test/intltest/usettest.cpp

```diff
@@ -4334,3 +4335,63 @@ void UnicodeSetTest::TestElementIterator() {
     // begin() & end() return USetElementIterator for which explicit APIs are tested via USet
     // in a header-only unit test file.
 }
+
+void UnicodeSetTest::TestParseErrors() {
+    for (const auto expression : std::vector<std::u16string_view>{
+             // Java error message: "Char expected after operator".
+             u"[a-[b]]",
+             // "Missing '['".
+             u"a-z",
+             // "Trailing '&'".
+             u"[[a]&]",
+             // "'-' not after char or set".
+             u"[[a]&-[z]]",
+             u"[[a]--[z]]",
+             u"[{aa}-{zz}]",
+             // "'&' not after set".
+             u"[a&z]",
+             u"[{aa}&{zz}]",
+             // "'^' not after '['"
+             u"[a^z]",  // TODO(egg): Exclude from literal-element in PDUTS61.
+             // "Missing operand after operator".
+             u"[a-{zz}]",
+             u"[[a]-{zz}]",
+             u"[[a]&{zz}]",
```

Comment:
FYI: It seems like this should have the same error message as `[a-[b]]` above...